resources/images/image.go

```diff
@@ -41,22 +41,62 @@ func (s *SoftwareImageMetaConstructor) Validate() error {
 	return err
 }
 
-// Information provided with YOCTO image
-type SoftwareImageMetaYoctoConstructor struct {
-	// Yocto ID build in the image
-	YoctoId string `json:"yocto_id" valid:"length(1|4096),required"`
+// Structure with artifact version informations
+type ArtifactInfo struct {
+	// Mender artifact format - the only possible value is "mender"
+	//Format string `json:"format" valid:"string,equal("mender"),required"`
+	Format string `json:"format" valid:"required"`
+
+	// Mender artifact format version
+	//Version uint `json:"version" valid:"uint,equal(1),required"`
+	Version uint `json:"version" valid:"required"`
+}
 
-	// Compatible device model for the application
-	DeviceType string `json:"device_type" bson:"device_type" valid:"length(1|4096),required"`
+// Type info structure
+type ArtifactUpdateTypeInfo struct {
+	Type string `json:"type" valid:"required"`
+}
+
+// Update file structure
+type UpdateFile struct {
+	// Image name
+	Name string `json:"name" valid:"required"`
 
 	// Image file checksum
-	Checksum string `json:"checksum" valid:"required"`
+	Checksum string `json:"checksum" valid:"optional"`
+
+	// Image file signature
+	Signature string `json:"signature" valid:"optional"`
 
 	// Image size
-	ImageSize int64 `json:"image_size" valid:"optional"`
+	Size int64 `json:"size" valid:"optional"`
 
 	// Date build
-	DateBuild *time.Time `json:"date_build" valid:"optional"`
+	Date *time.Time `json:"date" valid:"optional"`
+}
+
+type UpdateFiles []UpdateFile
+
+// Update structure
+type Update struct {
+	TypeInfo ArtifactUpdateTypeInfo `json:"type_info" valid:"required"`
+	Files    UpdateFiles            `json:"files"`
```

Comment:
`[]UpdateFile` seems fine to me, at least it's clearly visible this field is a list of `UpdateFile`.